components/server-side-render/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies.
+ */
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: null,
+		};
+	}
+
+	componentDidMount() {
+		this.fetch( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps, this.props ) ) {
+			this.fetch( nextProps );
+		}
+	}
+
+	fetch( props ) {
+		this.setState( { response: null } );
+		const { block } = props;
+		const attributes = Object.assign( {}, props );
+
+		// Delete 'block' from attributes, only registered block attributes are allowed.
+		delete attributes.block;
```

Comment:
Maybe instead of this this the component itself should take an `attributes` prop? So then the component could be rendered via:

```jsx
<ServerSideRender
	block="core/archives"
	attributes={ this.props.attributes }
	/>
```

This would allow the rendering of a block that itself has a `block` attribute.

So then there would be no need either for:

```js
const attributes = Object.assign( {}, props );
```

Below it can just do:

```js
attributes: props.attributes
```

Reply:
Replaced in https://github.com/WordPress/gutenberg/pull/5602/commits/e0fec83b254732ede34a69e54eaf531b86360528, definitely makes more sense, thank you!